### PR TITLE
npm: prevent scoped registries becoming global

### PR DESCRIPTION
**Pull request by feelepxyz**
Fixes a bug where all packages (public and private) are forced through a
private registry, where it's only intended to be used for scoped
packages so fails for public packages hosted on npm.

Added a check to the method that attempts to extract a global registry,
checking if credentials registry host has already been configured in
`.npmrc` as a scoped registry.

This prevents dependabot from re-writing the `.npmrc` global `registry`
with the scoped registry:

```
registry=registry.npmjs.org
@Private=private.example.com
registry=private.example.com
auth..
```

Keeping the global and scoped config intact:

```
registry=registry.npmjs.org
@Private=private.example.com
auth..
```

### CODE REVIEW

File: npm_and_yarn/spec/fixtures/projects/yarn/scoped_private_source_with_npmrc copy/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (brrygrdn) [inline npm_and_yarn/spec/fixtures/projects/yarn/scoped_private_source_with_npmrc copy/package.json:1]: Unintentional check-in?

File: npm_and_yarn/spec/fixtures/projects/yarn/scoped_private_source_with_npmrc copy/yarn.lock [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (brrygrdn) [inline npm_and_yarn/spec/fixtures/projects/yarn/scoped_private_source_with_npmrc copy/yarn.lock:1]: Same as above?

**Review by brrygrdn — CHANGES_REQUESTED** [review]
Change looks good to me, one question about the `foo copy` files I think might be cruft, otherwise 👍🏻

File: npm_and_yarn/spec/fixtures/projects/yarn/scoped_private_source_with_npmrc copy/package.json (hunk 1 above)
Comment (feelepxyz) [reply npm_and_yarn/spec/fixtures/projects/yarn/scoped_private_source_with_npmrc copy/package.json]: Yep good catch!

**Review by jurre — APPROVED** [review]
Thanks for tackling this

File: npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npmrc_builder.rb [hunk 3]
```diff
@@ -150,6 +153,16 @@ def credential_lines_for_npmrc
           ["always-auth = true"] + lines
         end
 
+        def npmrc_scoped_registries
+          return [] unless npmrc_file
+
+          @npmrc_scoped_registries ||=
+            npmrc_file.content.lines.select { |line| line.match?(/^\s*@/) && line.include?(":registry") }.
+            map do |line|
+              line.match(/:registry\s*=\s*(?<registry>\S+)/)&.named_captures&.fetch("registry")
```
Comment (jurre) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npmrc_builder.rb:162]: We could maybe extract a constant for the regex here, but not blocking for review

**Review by jurre — APPROVED** [review]

**Review by brrygrdn — APPROVED** [review]

**Pull request merged**